src/pal/src/include/pal/sharedmemory.h

```diff
@@ -27,18 +27,30 @@
 #define SHARED_MEMORY_MAX_FILE_NAME_CHAR_COUNT (_MAX_FNAME - 1)
 #define SHARED_MEMORY_MAX_NAME_CHAR_COUNT (_countof("Global\\") - 1 + SHARED_MEMORY_MAX_FILE_NAME_CHAR_COUNT)
 
+#ifndef __ANDROID__
 #define SHARED_MEMORY_TEMP_DIRECTORY_PATH "/tmp"
 #define SHARED_MEMORY_RUNTIME_TEMP_DIRECTORY_PATH "/tmp/.dotnet"
 
 #define SHARED_MEMORY_SHARED_MEMORY_DIRECTORY_PATH "/tmp/.dotnet/shm"
 #define SHARED_MEMORY_LOCK_FILES_DIRECTORY_PATH "/tmp/.dotnet/lockfiles"
+
+#define SHARED_MEMORY_UNIQUE_TEMP_NAME_TEMPLATE "/tmp/.coreclr.XXXXXX"
+#else
+#define SHARED_MEMORY_TEMP_DIRECTORY_PATH "/data/local/tmp"
```

Comment:
Maybe we can define somewhere a symbol TEMP_DIRECTORY_PATH conditionally to /tmp or /data/local/tmp based on whether we are on Android or not and then use it everywhere. So e.g.
```C++
#define SHARED_MEMORY_RUNTIME_TEMP_DIRECTORY_PATH TEMP_DIRECTORY_PATH "/.dotnet"
```
Then the #ifdef would be just at one place.